src/Symfony/Component/HttpKernel/Debug/FileLinkFormatter.php

```diff
@@ -22,22 +22,23 @@
 class FileLinkFormatter implements \Serializable
 {
     private $fileLinkFormat;
-    private $requestStack;
+    private $router;
     private $baseDir;
-    private $urlFormat;
+    private $queryString;
 
-    public function __construct($fileLinkFormat = null, RequestStack $requestStack = null, $baseDir = null, $urlFormat = null)
+    public function __construct($fileLinkFormat = null, RequestStack $requestStack = null, $baseDir = null, $queryString = null, UrlGeneratorInterface $router = null)
     {
+        // TODO: '$requestStack' constructor argument is unused. Remove it in Symfony 4.0.
```

Comment:
We won't be able to do that. What would be better is to remove the type hint for `$requestStack`, accept both a RequestStack and a UrlGeneratorInterface, trigger a deprecation if this is a RequestStack. That way, we can remove the deprecated part in 4.0.